### PR TITLE
Prepare auth

### PR DESCRIPTION
**Pull request by rlustin**
I’ve slitted the authentication work in many PR to facilitate the review.

This PR prepares the arrival of other controllers:
- Test env var on install and test script;
- Get parameters in a helper common to API controllers;
- Use Result error handling pattern for database connection;
- Set sqlite busy timeout from env variable.

@Signez, could you please review?

### CODE REVIEW

File: src/api/v1/translations.rs [hunk 1]
```diff
@@ -53,35 +47,21 @@ pub fn translations_index(_: &mut Request) -> IronResult<Response> {
...
-    let new_locale = match parameters.find(&["locale"]) {
-        Some(&Value::String(ref new_locale)) => new_locale,
-        _ => return bad_request_error("You must provide a locale"),
-    };
-
-    let new_content = match parameters.find(&["content"]) {
-        Some(&Value::String(ref new_content)) => new_content,
-        _ => return bad_request_error("You must provide a content"),
-    };
+    let new_key = try!(get_param(request, "key"));
```
Comment (Signez) [inline src/api/v1/translations.rs]: Miam! Great _near-zero cost abstraction_ here 👍

File: src/database.rs [hunk 2]
```diff
@@ -1,14 +1,19 @@
...
     dotenv().ok();
 
     let database_url = env::var("DATABASE_URL")
         .expect("DATABASE_URL must be set");
 
-    SqliteConnection::establish(&database_url)
-        .expect(&format!("Error connecting to {}", database_url))
+    let connection = try!(SqliteConnection::establish(&database_url));
+
+    let busy_timeout = env::var("DATABASE_BUSY_TIMEOUT").unwrap_or("250".to_string());
```
Comment (Signez) [inline src/database.rs]: Why not grouping it with the `DATABASE_URL` gathering earlier in the function?

Comment (rlustin) [inline src/database.rs]: Sure.

File: src/api/v1/translations.rs (hunk 1 above)
Comment (rlustin) [inline src/api/v1/translations.rs]: :)

**Comment by rlustin** [comment]
Fixed.

**Pull request merged**